src/main/kotlin/net/irisshaders/lilybot/extensions/util/ThreadControl.kt

```diff
@@ -151,8 +142,27 @@ class ThreadControl : Extension() {
 	inner class ThreadArchiveArgs : Arguments() {
 		val lock by defaultingBoolean {
 			name = "lock"
-			description = "Whether to lock this thread, if you are a moderator. Default is false"
+			description = "Whether to lock the thread if you are a moderator. Default is false"
 			defaultValue = false
 		}
 	}
+
+	inner class ThreadTransferArgs : Arguments() {
+		val newOwner by member {
+			name = "newOwner"
+			description = "The user you want to transfer ownership of the thread to"
+		}
+	}
+
+	private suspend fun EphemeralSlashCommandContext<*>.ownsThreadOrModerator
+				(inputThread: ThreadChannel, inputMember: Member): Boolean {
+		val databaseThreadOwner = DatabaseHelper.getThreadOwner(inputThread.id)
+
+		if (inputMember.hasPermission(Permission.ModerateMembers) || databaseThreadOwner == inputMember.id) {
```

Comment:
Are the curly braces necessary?

Reply:
DeteKt yells at me if I don't have them, so yes.

Reply:
detekt can start coping